### DISCUSSION TITLE
Why is Field not sealed?

### DISCUSSION THREAD

**Discussion by kubukoz** [SystemFw/dynosaur#21] [Q&A] [answered]
Hi! I was trying to write a utility for "nesting" a `Field[A, B]` inside a structure that contains an `A` - sort of a `contramap` thing. I noticed Field isn't sealed - is there any specific reason? It seems like the implementation relies on Required/Optional being the only values.

**Comment by kubukoz** [comment]
In general what I'm actually trying to provide is this:

```scala
def nestFieldK[A, S](
    f: S => A
  ): Field[A, *] ~> Field[S, *]
```

if it makes sense, I'd happily add it here, along with a `contramap` for fields that this would use underneath.

**Reply by SystemFw** [reply, marked as answer]
Obscenely late for an answer, but it's `sealed` now
